client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,215 @@
+				return Object.assign( {}, activeFilter, { [ property ]: value } );
+			}
+			return activeFilter;
+		} );
+
+		this.setState( { activeFilters } );
+	}
+
+	removeFilter( key ) {
+		const activeFilters = [ ...this.state.activeFilters ];
+		const index = findIndex( activeFilters, f => f.key === key );
+		activeFilters.splice( index, 1 );
+		this.setState( { activeFilters } );
+	}
+
+	getTitle() {
+		const { config } = this.props;
+		const { match } = this.state;
+		const label = sprintf( __( 'Orders Match %s Filters', 'wc-admin' ), match.label );
+		return (
+			<Fragment>
+				<span className="screen-reader-text">{ label }</span>
+				<span>{ __( 'Orders Match', 'wc-admin' ) }</span>
+				<SelectControl
+					className="woocommerce-advanced-filters__title-select"
+					options={ config.matches }
+					value={ match.value }
+					onChange={ this.onMatchChange }
+				/>
+				<span>{ __( 'Filters', 'wc-admin' ) }</span>
+			</Fragment>
+		);
+	}
+
+	getSelector( filter ) {
+		const filterConfig = this.props.config.filters[ filter.key ];
+		const { input } = filterConfig;
+		if ( ! input ) {
+			return null;
+		}
+		if ( 'SelectControl' === input.component ) {
+			return (
+				<SelectControl
+					className="woocommerce-advanced-filters__list-select"
+					options={ input.options }
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+				/>
+			);
+		}
+		if ( 'FormTokenField' === input.component ) {
+			return (
+				<FormTokenField
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+				/>
+			);
+		}
+		return null;
+	}
+
+	getAvailableFilterKeys() {
+		const { config } = this.props;
+		const activeFilterKeys = this.state.activeFilters.map( f => f.key );
+		return difference( Object.keys( config.filters ), activeFilterKeys );
+	}
+
+	addFilter( key, onClose ) {
+		const filterConfig = this.props.config.filters[ key ];
+		const newFilter = { key, rule: filterConfig.rules[ 0 ] };
+		if ( filterConfig.input && filterConfig.input.options ) {
+			newFilter.value = filterConfig.input.options[ 0 ];
+		}
+		this.setState( state => {
+			return {
+				activeFilters: [ ...state.activeFilters, newFilter ],
+			};
+		} );
+		onClose();
+	}
+
+	clearAll() {
+		this.setState( {
+			activeFilters: [],
+		} );
+	}
+
+	render() {
+		const { config } = this.props;
+		const availableFilterKeys = this.getAvailableFilterKeys();
+		return (
+			<Card className="woocommerce-advanced-filters" title={ this.getTitle() }>
+				<ul className="woocommerce-advanced-filters__list">
+					{ this.state.activeFilters.map( filter => {
+						const { key, rule } = filter;
+						const filterConfig = config.filters[ key ];
+						return (
+							<li className="woocommerce-advanced-filters__list-item" key={ key }>
+								<fieldset>
+									<div className="woocommerce-advanced-filters__fieldset">
+										<legend>{ filterConfig.label }</legend>
```

Comment:
The `legend` needs to be the first child of `fieldset` in order for it to label the contents

Reply:
TIL